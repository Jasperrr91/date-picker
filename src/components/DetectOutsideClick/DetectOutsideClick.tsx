import React, { useRef, useEffect } from 'react';

type Props = {
  handleClick: () => void;
  children: React.ReactNode;
};

const useOutsideClick = (handleClick: any, ref: any) => {
  useEffect(() => {
    const outsideClick = (event: any) => {
      if (ref.current && !ref.current.contains(event.target)) {
        handleClick();
      }
    };

    document.addEventListener('click', outsideClick);

    return () => {
      document.removeEventListener('click', outsideClick);
    };
  }, [handleClick, ref]);
};

const DetectOutsideClick = ({ handleClick, children }: Props) => {
  const wrapperRef = useRef(null);
  useOutsideClick(handleClick, wrapperRef);

  return (
    <div ref={wrapperRef}>
      {children}
    </div>
  );
};

export default DetectOutsideClick;
