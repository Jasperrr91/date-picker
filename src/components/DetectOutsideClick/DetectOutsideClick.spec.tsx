import React from 'react';
import { shallow } from 'enzyme';
import DetectOutsideClick from './DetectOutsideClick';

describe('DetectOutsideClick', () => {
  it('It will render a DetectOutsideClick component correctly', () => {
    const handleClick = jest.fn();
    const app = shallow(<DetectOutsideClick handleClick={handleClick}>child</DetectOutsideClick>);
    expect(app).toMatchSnapshot();
  });
});
