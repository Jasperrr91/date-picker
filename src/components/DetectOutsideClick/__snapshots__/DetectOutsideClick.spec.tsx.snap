// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetectOutsideClick It will render a DetectOutsideClick component correctly 1`] = `
<div>
  child
</div>
`;
